diff --git a/chapter-03-01-state-starter/src/root.tsx b/chapter-03-02-new-component/src/root.tsx
--- a/chapter-03-01-state-starter/src/root.tsx
+++ b/chapter-03-02-new-component/src/root.tsx
@@ -1,27 +1,33 @@
-import { repoList } from "./data";
+import { repoList, type TRepoSearchResultItem } from "./data";
 import { formatRelativeTime } from "./helpers/time";
 
 export function GitHubRoot() {
   return (
     <div className="flex flex-col gap-6">
       {repoList.map((item) => {
-        const formattedCreatedAt = formatRelativeTime(item.created_at);
-
-        return (
-          <div key={item.id}>
-            <h2 className="font-medium">
-              <a className="hover:underline" href={item.html_url}>
-                {item.full_name}
-              </a>
-            </h2>
-            <p>{item.description}</p>
-            <p className="text-sm text-gray-600">
-              {item.stargazers_count.toLocaleString("en-US")} stars | created{" "}
-              {formattedCreatedAt}
-            </p>
-          </div>
-        );
+        return <RepoItem key={item.id} data={item} />;
       })}
     </div>
   );
 }
+
+function RepoItem(props: { data: TRepoSearchResultItem }) {
+  const { data } = props;
+
+  const formattedCreatedAt = formatRelativeTime(data.created_at);
+
+  return (
+    <div>
+      <h2 className="font-medium">
+        <a className="hover:underline" href={data.html_url}>
+          {data.full_name}
+        </a>
+      </h2>
+      <p>{data.description}</p>
+      <p className="text-sm text-gray-600">
+        {data.stargazers_count.toLocaleString("en-US")} stars | created{" "}
+        {formattedCreatedAt}
+      </p>
+    </div>
+  );
+}
