diff --git a/chapter-01-02-just-titles/src/root.tsx b/chapter-01-03-first-map/src/root.tsx
--- a/chapter-01-02-just-titles/src/root.tsx
+++ b/chapter-01-03-first-map/src/root.tsx
@@ -1,13 +1,18 @@
-export function GitHubRoot() {
-  const name1 = "facebook/react";
-  const name2 = "facebook/react-native";
-  const name3 = "vercel/next.js";
+const data = [
+  { id: 10270250, name: "facebook/react" },
+  { id: 29028775, name: "facebook/react-native" },
+  { id: 70107786, name: "vercel/next.js" },
+  { id: 135786093, name: "typescript-cheatsheets/react" },
+  { id: 174038031, name: "react-hook-form/react-hook-form" },
+  { id: 585146387, name: "shadcn-ui/ui" },
+];
 
+export function GitHubRoot() {
   return (
     <div>
-      <p>{name1}</p>
-      <p>{name2}</p>
-      <p>{name3}</p>
+      {data.map((item) => {
+        return <p key={item.id}>{item.name}</p>;
+      })}
     </div>
   );
 }
