diff --git a/chapter-03-07-pagination-button/src/root.tsx b/chapter-03-08-remember-unique-key/src/root.tsx
--- a/chapter-03-07-pagination-button/src/root.tsx
+++ b/chapter-03-08-remember-unique-key/src/root.tsx
@@ -18,8 +18,8 @@ export function GitHubRoot() {
   return (
     <div className="p-8">
       <div className="flex flex-col gap-6">
-        {visibleWindow.map((item) => {
-          return <NewsItem key={item.id} data={item} />;
+        {visibleWindow.map((item, index) => {
+          return <NewsItem key={index} data={item} />;
         })}
       </div>
 
