diff --git a/chapter-03-09-derived-value/src/root.tsx b/chapter-03-10-bonus-conditional-styles/src/root.tsx
index b15873e..e316b7a 100644
--- a/chapter-03-09-derived-value/src/root.tsx
+++ b/chapter-03-10-bonus-conditional-styles/src/root.tsx
@@ -1,4 +1,5 @@
 import * as React from "react";
+import cx from "classnames";
 import { storyList, type TStoryItem } from "./data";
 import { formatRelativeTime } from "./helpers/time";
 
@@ -29,7 +30,10 @@ export function HackerNews() {
       <div className="mt-12">
         <button
           disabled={!canGoBack}
-          className="hover:underline"
+          className={cx({
+            "hover:underline": canGoBack,
+            "text-gray-300": !canGoBack,
+          })}
           onClick={() => {
             setCurrentPageNumber((n) => n - 1);
           }}
@@ -43,7 +47,10 @@ export function HackerNews() {
 
         <button
           disabled={!canGoNext}
-          className="hover:underline"
+          className={cx({
+            "hover:underline": canGoNext,
+            "text-gray-300": !canGoNext,
+          })}
           onClick={() => {
             setCurrentPageNumber((n) => n + 1);
           }}
