diff --git a/chapter-01-04-real-data/src/root.tsx b/chapter-01-05-pre-trick/src/root.tsx
index be5b9d1..5d76737 100644
--- a/chapter-01-04-real-data/src/root.tsx
+++ b/chapter-01-05-pre-trick/src/root.tsx
@@ -4,7 +4,7 @@ export function GitHubRoot() {
   return (
     <div>
       {data.map((item) => {
-        return <p key={item.id}>{item.title}</p>;
+        return <pre key={item.id}>{JSON.stringify(item, null, 2)}</pre>;
       })}
     </div>
   );
