diff --git a/chapter-01-09-basic-layout/src/root.tsx b/chapter-01-10-more-styling/src/root.tsx
index f4e632e..cda7777 100644
--- a/chapter-01-09-basic-layout/src/root.tsx
+++ b/chapter-01-10-more-styling/src/root.tsx
@@ -20,7 +20,7 @@ export function GitHubRoot() {
             <p>
               <a href={item.url}>{item.title}</a>
             </p>
-            <p>
+            <p className="text-sm text-gray-600">
               {item.score} points by {item.by} {formattedTime} |{" "}
               {item.descendants} comments
             </p>
