diff --git a/chapter-04-09-reset-pagination-state/src/root.tsx b/chapter-04-10-prefetch-data/src/root.tsx
--- a/chapter-04-09-reset-pagination-state/src/root.tsx
+++ b/chapter-04-10-prefetch-data/src/root.tsx
@@ -117,10 +117,25 @@ export function SearchResult(props: { searchQ: string }) {
         }}
         totalPageCount={totalPageCount}
       />
+
+      {!canGoForward ? null : (
+        <PrefetchPage searchQ={searchQ} pageNumber={currentPageNumber + 2} />
+      )}
     </>
   );
 }
 
+function PrefetchPage(props: { searchQ: string; pageNumber: number }) {
+  const { searchQ, pageNumber } = props;
+
+  useGitHubRepoSearchQuery({
+    q: searchQ,
+    pageNumber,
+  });
+
+  return null;
+}
+
 function useGitHubRepoLatestReleaseQuery(input: {
   owner: string;
   repo: string;
