diff --git a/chapter-03-04-stale-state-bug/src/root.tsx b/chapter-03-05-state-functional-updater/src/root.tsx
--- a/chapter-03-04-stale-state-bug/src/root.tsx
+++ b/chapter-03-05-state-functional-updater/src/root.tsx
@@ -42,9 +42,9 @@ function NewsItem(props: { data: TStoryItem }) {
         <button
           className="hover:underline"
           onClick={() => {
-            setLikeCount(likeCount + 1);
-            setLikeCount(likeCount + 1);
-            setLikeCount(likeCount + 1);
+            setLikeCount((num) => num + 1);
+            setLikeCount((num) => num + 1);
+            setLikeCount((num) => num + 1);
           }}
         >
           like ({likeCount})
