diff --git a/chapter-01-08-with-date-fns/src/root.tsx b/chapter-01-09-basic-layout/src/root.tsx
--- a/chapter-01-08-with-date-fns/src/root.tsx
+++ b/chapter-01-09-basic-layout/src/root.tsx
@@ -4,7 +4,7 @@ import data from "./data.json";
 
 export function GitHubRoot() {
   return (
-    <div>
+    <div className="flex flex-col gap-6 p-8">
       {data.map((item) => {
         // See https://date-fns.org/v2.30.0/docs/formatDistanceToNowStrict
         const formattedCreatedAt = formatDistanceToNowStrict(
