diff --git a/chapter-01-11-quiz-answers/src/root.tsx b/chapter-03-01-state-starter/src/root.tsx
index 8c3914e..5ec9b60 100644
--- a/chapter-01-11-quiz-answers/src/root.tsx
+++ b/chapter-03-01-state-starter/src/root.tsx
@@ -1,39 +1,35 @@
-import formatDistanceToNowStrict from "date-fns/formatDistanceToNowStrict";
-import { enUS as localeEnUs } from "date-fns/locale";
-import data from "./data.json";
+import { storyList } from "./data";
+import { formatRelativeTime } from "./helpers/time";
 
 export function GitHubRoot() {
   return (
-    <div className="flex flex-col gap-6 p-8">
-      {data.map((item) => {
-        // See https://date-fns.org/v2.30.0/docs/formatDistanceToNowStrict
-        const formattedTime = formatDistanceToNowStrict(
-          item.time * 1000, // seconds -> milliseconds
-          {
-            addSuffix: true, // add "ago"
-            locale: localeEnUs,
-          },
-        );
+    <div className="p-8">
+      <div className="flex flex-col gap-6">
+        {storyList.map((item) => {
+          const formattedTime = formatRelativeTime(item.time);
 
-        return (
-          <div key={item.id}>
-            <p>
-              <a className="hover:underline" href={item.url}>
-                {item.title}
-              </a>
-            </p>
-            <p className="text-sm text-gray-600">
-              {item.score} points by {item.by} {formattedTime} |{" "}
-              <a
-                className="hover:underline"
-                href={`https://news.ycombinator.com/item?id=${item.id}`}
-              >
-                {item.descendants} comments
-              </a>
-            </p>
-          </div>
-        );
-      })}
+          return (
+            <div key={item.id}>
+              <p>
+                <a className="hover:underline" href={item.url}>
+                  {item.title}
+                </a>
+              </p>
+
+              <p className="text-sm text-gray-600">
+                {item.score} points by {item.by} {formattedTime}
+                {" | "}
+                <a
+                  className="hover:underline"
+                  href={`https://news.ycombinator.com/item?id=${item.id}`}
+                >
+                  {item.descendants} comments
+                </a>
+              </p>
+            </div>
+          );
+        })}
+      </div>
     </div>
   );
 }
