diff --git a/chapter-03-07-pagination-button/src/root.tsx b/chapter-03-09-derived-value/src/root.tsx
--- a/chapter-03-07-pagination-button/src/root.tsx
+++ b/chapter-03-09-derived-value/src/root.tsx
@@ -13,6 +13,9 @@ export function GitHubRoot() {
   const startIndex = currentPageNumber * PAGE_SIZE; // Inclusive
   const endIndex = (currentPageNumber + 1) * PAGE_SIZE; // Exclusive
 
+  const canGoBack = startIndex > 0;
+  const canGoForward = endIndex < totalItemCount;
+
   const visibleWindow = repoList.slice(startIndex, endIndex);
 
   return (
@@ -25,6 +28,7 @@ export function GitHubRoot() {
 
       <div className="mt-12">
         <button
+          disabled={!canGoBack}
           className="hover:underline"
           onClick={() => {
             setCurrentPageNumber((n) => n - 1);
@@ -38,6 +42,7 @@ export function GitHubRoot() {
         </span>
 
         <button
+          disabled={!canGoForward}
           className="hover:underline"
           onClick={() => {
             setCurrentPageNumber((n) => n + 1);
