diff --git a/chapter-04-06-decouple-input-and-query-states/src/root.tsx b/chapter-04-07-enter-to-submit/src/root.tsx
--- a/chapter-04-06-decouple-input-and-query-states/src/root.tsx
+++ b/chapter-04-07-enter-to-submit/src/root.tsx
@@ -15,7 +15,14 @@ export function GitHubRoot() {
 
   return (
     <>
-      <div className="mb-6 flex gap-4">
+      <form
+        className="mb-6 flex gap-4"
+        onSubmit={(event) => {
+          event.preventDefault();
+
+          setSearchQ(rawInput);
+        }}
+      >
         <input
           className="w-60 rounded-md border px-3 py-1 text-sm shadow-sm"
           onChange={(event) => {
@@ -28,13 +35,11 @@ export function GitHubRoot() {
 
         <button
           className="rounded-md border px-3 py-1 text-sm font-medium shadow-sm active:scale-95"
-          onClick={() => {
-            setSearchQ(rawInput);
-          }}
+          type="submit"
         >
           Search
         </button>
-      </div>
+      </form>
 
       <SearchResult searchQ={searchQ} />
     </>
