diff --git a/chapter-03-10-bonus-conditional-styles/src/root.tsx b/chapter-03-11-lift-state-up/src/root.tsx
--- a/chapter-03-10-bonus-conditional-styles/src/root.tsx
+++ b/chapter-03-11-lift-state-up/src/root.tsx
@@ -7,6 +7,9 @@ const PAGE_SIZE = 8;
 
 export function GitHubRoot() {
   const [currentPageNumber, setCurrentPageNumber] = React.useState<number>(0);
+  const [likeCountMap, setLikeCountMap] = React.useState<Map<number, number>>(
+    new Map(),
+  );
 
   const totalItemCount = repoList.length;
   const totalPageCount = Math.ceil(totalItemCount / PAGE_SIZE);
@@ -23,7 +26,22 @@ export function GitHubRoot() {
     <>
       <div className="flex flex-col gap-6">
         {visibleWindow.map((item) => {
-          return <RepoItem key={item.id} data={item} />;
+          return (
+            <RepoItem
+              key={item.id}
+              data={item}
+              likeCount={likeCountMap.get(item.id) ?? 0}
+              onLikeClick={() => {
+                setLikeCountMap((map) => {
+                  const copy = new Map(map);
+                  const currentLikeCount = copy.get(item.id) ?? 0;
+
+                  copy.set(item.id, currentLikeCount + 1);
+                  return copy;
+                });
+              }}
+            />
+          );
         })}
       </div>
 
@@ -62,10 +80,12 @@ export function GitHubRoot() {
   );
 }
 
-function RepoItem(props: { data: TRepoSearchResultItem }) {
-  const { data } = props;
-
-  const [likeCount, setLikeCount] = React.useState<number>(0);
+function RepoItem(props: {
+  data: TRepoSearchResultItem;
+  likeCount: number;
+  onLikeClick: () => void;
+}) {
+  const { data, likeCount, onLikeClick } = props;
 
   const formattedCreatedAt = formatRelativeTime(data.created_at);
 
@@ -84,7 +104,7 @@ function RepoItem(props: { data: TRepoSearchResultItem }) {
         <button
           className="hover:underline"
           onClick={() => {
-            setLikeCount((num) => num + 1);
+            onLikeClick();
           }}
         >
           like ({likeCount})
