diff --git a/chapter-01-04-real-data/src/root.tsx b/chapter-01-05-pre-trick/src/root.tsx
--- a/chapter-01-04-real-data/src/root.tsx
+++ b/chapter-01-05-pre-trick/src/root.tsx
@@ -4,7 +4,7 @@ export function GitHubRoot() {
   return (
     <div>
       {repoList.map((item) => {
-        return <h2 key={item.id}>{item.full_name}</h2>;
+        return <pre key={item.id}>{JSON.stringify(item, null, 2)}</pre>;
       })}
     </div>
   );
