diff --git a/chapter-01-03-first-map/src/root.tsx b/chapter-01-04-real-data/src/root.tsx
--- a/chapter-01-03-first-map/src/root.tsx
+++ b/chapter-01-04-real-data/src/root.tsx
@@ -1,17 +1,10 @@
-const data = [
-  { id: 10270250, name: "facebook/react" },
-  { id: 29028775, name: "facebook/react-native" },
-  { id: 70107786, name: "vercel/next.js" },
-  { id: 135786093, name: "typescript-cheatsheets/react" },
-  { id: 174038031, name: "react-hook-form/react-hook-form" },
-  { id: 585146387, name: "shadcn-ui/ui" },
-];
+import data from "./data.json";
 
 export function GitHubRoot() {
   return (
     <div>
       {data.map((item) => {
-        return <p key={item.id}>{item.name}</p>;
+        return <p key={item.id}>{item.full_name}</p>;
       })}
     </div>
   );
