diff --git a/chapter-01-10-more-styling/src/root.tsx b/chapter-01-11-quiz-answers/src/root.tsx
--- a/chapter-01-10-more-styling/src/root.tsx
+++ b/chapter-01-11-quiz-answers/src/root.tsx
@@ -18,11 +18,20 @@ export function GitHubRoot() {
         return (
           <div key={item.id}>
             <h2 className="font-medium">
-              <a href={item.html_url}>{item.full_name}</a>
+              <a className="hover:underline" href={item.html_url}>
+                {item.full_name}
+              </a>
             </h2>
             <p>{item.description}</p>
             <p className="text-sm text-gray-600">
-              {item.stargazers_count} stars | created {formattedCreatedAt}
+              <a
+                className="hover:underline"
+                href={`https://github.com/${item.full_name}/stargazers`}
+              >
+                {item.stargazers_count.toLocaleString("en-US")} stars
+              </a>
+              {" | "}
+              created {formattedCreatedAt}
             </p>
           </div>
         );
