diff --git a/chapter-04-04-dynamic-query/src/root.tsx b/chapter-04-05-search-input/src/root.tsx
--- a/chapter-04-04-dynamic-query/src/root.tsx
+++ b/chapter-04-05-search-input/src/root.tsx
@@ -9,6 +9,26 @@ const octokit = new Octokit({
   auth: process.env.NEXT_PUBLIC_UNSAFE_LOCAL_ONLY_GITHUB_PERSONAL_ACCESS_TOKEN,
 });
 
+export function GitHubRoot() {
+  const [searchQ, setSearchQ] = React.useState("react+stars:>100");
+
+  return (
+    <>
+      <div className="mb-6">
+        <input
+          className="rounded-md border px-3 py-1 text-sm shadow-sm"
+          onChange={(event) => {
+            setSearchQ(event.currentTarget.value);
+          }}
+          value={searchQ}
+        />
+      </div>
+
+      <SearchResult searchQ={searchQ} />
+    </>
+  );
+}
+
 function useGitHubRepoSearchQuery(q: string) {
   const cacheKey = ["octokit.search.repos", q];
 
@@ -25,8 +45,10 @@ function useGitHubRepoSearchQuery(q: string) {
   });
 }
 
-export function GitHubRoot() {
-  const query = useGitHubRepoSearchQuery("react+stars:>100");
+export function SearchResult(props: { searchQ: string }) {
+  const { searchQ } = props;
+
+  const query = useGitHubRepoSearchQuery(searchQ);
   const response = query.data;
 
   if (query.isLoading) {
