diff --git a/chapter-01-01-hello-world/src/root.tsx b/chapter-01-02-just-titles/src/root.tsx
--- a/chapter-01-01-hello-world/src/root.tsx
+++ b/chapter-01-02-just-titles/src/root.tsx
@@ -1,3 +1,13 @@
-export function Hello() {
-  return <p>Hello Condensed!</p>;
+export function GitHubRoot() {
+  const title1 = "How Is LLaMa.cpp Possible?";
+  const title2 = "NetMaker: Connect Everything with a WireGuard VPN";
+  const title3 = "Opendream: A layer-based UI for Stable Diffusion";
+
+  return (
+    <div>
+      <p>{title1}</p>
+      <p>{title2}</p>
+      <p>{title3}</p>
+    </div>
+  );
 }
