diff --git a/chapter-01-06-more-ui/src/root.tsx b/chapter-01-07-adding-links/src/root.tsx
--- a/chapter-01-06-more-ui/src/root.tsx
+++ b/chapter-01-07-adding-links/src/root.tsx
@@ -6,7 +6,9 @@ export function GitHubRoot() {
       {data.map((item) => {
         return (
           <div key={item.id}>
-            <p>{item.full_name}</p>
+            <p>
+              <a href={item.html_url}>{item.full_name}</a>
+            </p>
             <p>{item.description}</p>
             <p>{item.stargazers_count} stars</p>
           </div>
