diff --git a/chapter-01-01-hello-world/src/root.tsx b/chapter-01-02-just-titles/src/root.tsx
--- a/chapter-01-01-hello-world/src/root.tsx
+++ b/chapter-01-02-just-titles/src/root.tsx
@@ -1,3 +1,13 @@
-export function Hello() {
-  return <p>Hello Condensed!</p>;
+export function GitHubRoot() {
+  const name1 = "facebook/react";
+  const name2 = "facebook/react-native";
+  const name3 = "vercel/next.js";
+
+  return (
+    <div>
+      <p>{name1}</p>
+      <p>{name2}</p>
+      <p>{name3}</p>
+    </div>
+  );
 }
