diff --git a/chapter-01-06-more-ui/src/root.tsx b/chapter-01-07-adding-links/src/root.tsx
index c2522d9..6726a51 100644
--- a/chapter-01-06-more-ui/src/root.tsx
+++ b/chapter-01-07-adding-links/src/root.tsx
@@ -6,7 +6,9 @@ export function HackerNews() {
       {data.map((item) => {
         return (
           <div key={item.id}>
-            <p>{item.title}</p>
+            <p>
+              <a href={item.url}>{item.title}</a>
+            </p>
             <p>
               {item.score} points by {item.by} | {item.descendants} comments
             </p>
