diff --git a/chapter-01-10-more-styling/src/root.tsx b/chapter-01-11-quiz-answers/src/root.tsx
--- a/chapter-01-10-more-styling/src/root.tsx
+++ b/chapter-01-11-quiz-answers/src/root.tsx
@@ -7,8 +7,8 @@ export function GitHubRoot() {
     <div className="flex flex-col gap-6 p-8">
       {data.map((item) => {
         // See https://date-fns.org/v2.30.0/docs/formatDistanceToNowStrict
-        const formattedCreatedAt = formatDistanceToNowStrict(
-          new Date(item.created_at), // parse ISO 8601 date time string
+        const formattedTime = formatDistanceToNowStrict(
+          item.time * 1000, // seconds -> milliseconds
           {
             addSuffix: true, // add "ago"
             locale: localeEnUs,
@@ -17,12 +17,19 @@ export function GitHubRoot() {
 
         return (
           <div key={item.id}>
-            <p className="font-medium">
-              <a href={item.html_url}>{item.full_name}</a>
+            <p>
+              <a className="hover:underline" href={item.url}>
+                {item.title}
+              </a>
             </p>
-            <p>{item.description}</p>
             <p className="text-sm text-gray-600">
-              {item.stargazers_count} stars | created {formattedCreatedAt}
+              {item.score} points by {item.by} {formattedTime} |{" "}
+              <a
+                className="hover:underline"
+                href={`https://news.ycombinator.com/item?id=${item.id}`}
+              >
+                {item.descendants} comments
+              </a>
             </p>
           </div>
         );
