diff --git a/chapter-04-03-component-colocated-query/src/root.tsx b/chapter-04-04-dynamic-query/src/root.tsx
--- a/chapter-04-03-component-colocated-query/src/root.tsx
+++ b/chapter-04-04-dynamic-query/src/root.tsx
@@ -10,7 +10,7 @@ const octokit = new Octokit({
 });
 
 export function GitHubRoot() {
-  const query = useGitHubRepoSearchQuery();
+  const query = useGitHubRepoSearchQuery("react+stars:>100");
   const response = query.data;
 
   if (query.isLoading) {
@@ -77,14 +77,14 @@ function RepoItem(props: { data: TRepoSearchResultItem }) {
   );
 }
 
-function useGitHubRepoSearchQuery() {
-  const cacheKey = ["octokit.search.repos"];
+function useGitHubRepoSearchQuery(q: string) {
+  const cacheKey = ["octokit.search.repos", q];
 
   return useSWR(cacheKey, async () => {
     // https://docs.github.com/en/search-github/searching-on-github/searching-for-repositories
     // https://docs.github.com/en/search-github/getting-started-with-searching-on-github/understanding-the-search-syntax
     return await octokit.search.repos({
-      q: "react+stars:>100",
+      q,
       sort: "stars",
       order: "desc",
       per_page: 24,
