diff --git a/chapter-03-01-state-starter/src/root.tsx b/chapter-03-02-new-component/src/root.tsx
index 269d012..0dd5d6e 100644
--- a/chapter-03-01-state-starter/src/root.tsx
+++ b/chapter-03-02-new-component/src/root.tsx
@@ -1,4 +1,4 @@
-import { storyList } from "./data";
+import { storyList, type TStoryItem } from "./data";
 import { formatRelativeTime } from "./helpers/time";
 
 export function HackerNews() {
@@ -6,30 +6,36 @@ export function HackerNews() {
     <div className="p-8">
       <div className="flex flex-col gap-6">
         {storyList.map((item) => {
-          const formattedTime = formatRelativeTime(item.time);
-
-          return (
-            <div key={item.id}>
-              <p>
-                <a className="hover:underline" href={item.url}>
-                  {item.title}
-                </a>
-              </p>
-
-              <p className="text-sm text-gray-600">
-                {item.score} points by {item.by} {formattedTime}
-                {" | "}
-                <a
-                  className="hover:underline"
-                  href={`https://news.ycombinator.com/item?id=${item.id}`}
-                >
-                  {item.descendants} comments
-                </a>
-              </p>
-            </div>
-          );
+          return <NewsItem key={item.id} data={item} />;
         })}
       </div>
     </div>
   );
 }
+
+function NewsItem(props: { data: TStoryItem }) {
+  const { data } = props;
+
+  const formattedTime = formatRelativeTime(data.time);
+
+  return (
+    <div>
+      <p>
+        <a className="hover:underline" href={data.url}>
+          {data.title}
+        </a>
+      </p>
+
+      <p className="text-sm text-gray-600">
+        {data.score} points by {data.by} {formattedTime}
+        {" | "}
+        <a
+          className="hover:underline"
+          href={`https://news.ycombinator.com/item?id=${data.id}`}
+        >
+          {data.descendants} comments
+        </a>
+      </p>
+    </div>
+  );
+}
