--- a/chapter-01-06-more-ui/src/root.tsx
+++ b/chapter-01-07-adding-links/src/root.tsx
@@ -6,7 +6,9 @@ export function GitHubRoot() {
       {repoList.map((item) => {
         return (
           <div key={item.id}>
-            <h2>{item.full_name}</h2>
+            <h2>
+              <a href={item.html_url}>{item.full_name}</a>
+            </h2>
             <p>{item.description}</p>
             <p>{item.stargazers_count} stars</p>
           </div>
