diff --git a/chapter-01-09-basic-layout/src/root.tsx b/chapter-01-10-more-styling/src/root.tsx
--- a/chapter-01-09-basic-layout/src/root.tsx
+++ b/chapter-01-10-more-styling/src/root.tsx
@@ -17,11 +17,11 @@ export function GitHubRoot() {
 
         return (
           <div key={item.id}>
-            <h2>
+            <h2 className="font-medium">
               <a href={item.html_url}>{item.full_name}</a>
             </h2>
             <p>{item.description}</p>
-            <p>
+            <p className="text-sm text-gray-600">
               {item.stargazers_count} stars | created {formattedCreatedAt}
             </p>
           </div>
