--- a/chapter-04-05-search-input/src/root.tsx
+++ b/chapter-04-06-decouple-input-and-query-states/src/root.tsx
@@ -10,20 +10,30 @@ const octokit = new Octokit({
 });
 
 export function GitHubRoot() {
+  const [rawInput, setRawInput] = React.useState<string>("react+stars:>100");
   const [searchQ, setSearchQ] = React.useState<string>("react+stars:>100");
 
   return (
     <>
-      <div className="mb-6">
+      <div className="mb-6 flex gap-4">
         <input
           className="w-60 rounded-md border px-3 py-1 text-sm shadow-sm"
           onChange={(event) => {
             const newInputValue = event.currentTarget.value;
 
-            setSearchQ(newInputValue);
+            setRawInput(newInputValue);
           }}
-          value={searchQ}
+          value={rawInput}
         />
+
+        <button
+          className="rounded-md border px-3 py-1 text-sm font-medium shadow-sm active:scale-95"
+          onClick={() => {
+            setSearchQ(rawInput);
+          }}
+        >
+          Search
+        </button>
       </div>
 
       <SearchResult searchQ={searchQ} />
