--- a/chapter-04-09-reset-pagination-state/src/root.tsx
+++ b/chapter-04-10-prefetch-data/src/root.tsx
@@ -100,6 +100,10 @@ function SearchResult(props: { searchQ: string }) {
         }}
         totalPageCount={totalPageCount}
       />
+
+      {canGoForward ? (
+        <PrefetchPage searchQ={searchQ} pageNumber={currentPageNumber + 2} />
+      ) : null}
     </>
   );
 }
@@ -145,6 +149,17 @@ function RepoItem(props: { data: TRepoSearchResultItem }) {
   );
 }
 
+function PrefetchPage(props: { searchQ: string; pageNumber: number }) {
+  const { searchQ, pageNumber } = props;
+
+  useGitHubRepoSearchQuery({
+    q: searchQ,
+    pageNumber,
+  });
+
+  return null;
+}
+
 function useGitHubRepoSearchQuery(input: { q: string; pageNumber: number }) {
   const { q, pageNumber } = input;
   const cacheKey = ["octokit.search.repos", q, pageNumber];
