--- a/chapter-03-05-state-functional-updater/src/root.tsx
+++ b/chapter-03-06-pagination-state/src/root.tsx
@@ -2,10 +2,19 @@ import * as React from "react";
 import { repoList, type TRepoSearchResultItem } from "./data";
 import { formatRelativeTime } from "./helpers/time";
 
+const PAGE_SIZE = 8;
+
 export function GitHubRoot() {
+  const [currentPageNumber, setCurrentPageNumber] = React.useState<number>(0);
+
+  const startIndex = currentPageNumber * PAGE_SIZE; // Inclusive
+  const endIndex = (currentPageNumber + 1) * PAGE_SIZE; // Exclusive
+
+  const visibleWindow = repoList.slice(startIndex, endIndex);
+
   return (
     <div className="flex flex-col gap-6">
-      {repoList.map((item) => {
+      {visibleWindow.map((item) => {
         return <RepoItem key={item.id} data={item} />;
       })}
     </div>
@@ -35,8 +44,6 @@ function RepoItem(props: { data: TRepoSearchResultItem }) {
           className="hover:underline"
           onClick={() => {
             setLikeCount((num) => num + 1);
-            setLikeCount((num) => num + 1);
-            setLikeCount((num) => num + 1);
           }}
         >
           like ({likeCount})
