--- a/chapter-03-11-lift-state-up/src/root.tsx
+++ b/chapter-04-01-data-fetching-starter/src/root.tsx
@@ -1,114 +1,36 @@
 import * as React from "react";
-import cx from "classnames";
 import { repoList, type TRepoSearchResultItem } from "./data";
 import { formatRelativeTime } from "./helpers/time";
 
-const PAGE_SIZE = 8;
-
 export function GitHubRoot() {
-  const [currentPageNumber, setCurrentPageNumber] = React.useState<number>(0);
-  const [likeCountMap, setLikeCountMap] = React.useState<Map<number, number>>(
-    new Map(),
-  );
-
-  const totalItemCount = repoList.length;
-  const totalPageCount = Math.ceil(totalItemCount / PAGE_SIZE);
-
-  const startIndex = currentPageNumber * PAGE_SIZE; // Inclusive
-  const endIndex = (currentPageNumber + 1) * PAGE_SIZE; // Exclusive
-
-  const canGoBack = startIndex > 0;
-  const canGoForward = endIndex < totalItemCount;
-
-  const visibleWindow = repoList.slice(startIndex, endIndex);
-
   return (
-    <>
-      <div className="flex flex-col gap-6">
-        {visibleWindow.map((item) => {
-          return (
-            <RepoItem
-              key={item.id}
-              data={item}
-              likeCount={likeCountMap.get(item.id) ?? 0}
-              onLikeClick={() => {
-                setLikeCountMap((map) => {
-                  const copy = new Map(map);
-                  const currentLikeCount = copy.get(item.id) ?? 0;
-
-                  copy.set(item.id, currentLikeCount + 1);
-                  return copy;
-                });
-              }}
-            />
-          );
-        })}
-      </div>
-
-      <div className="mt-12">
-        <button
-          disabled={!canGoBack}
-          className={cx({
-            "hover:underline": canGoBack,
-            "text-gray-300": !canGoBack,
-          })}
-          onClick={() => {
-            setCurrentPageNumber((n) => n - 1);
-          }}
-        >
-          Prev
-        </button>
-
-        <span className="mx-8">
-          {currentPageNumber + 1} / {totalPageCount}
-        </span>
-
-        <button
-          disabled={!canGoForward}
-          className={cx({
-            "hover:underline": canGoForward,
-            "text-gray-300": !canGoForward,
-          })}
-          onClick={() => {
-            setCurrentPageNumber((n) => n + 1);
-          }}
-        >
-          Next
-        </button>
-      </div>
-    </>
+    <div className="flex flex-col gap-6">
+      {repoList.map((item) => {
+        return <RepoItem key={item.id} data={item} />;
+      })}
+    </div>
   );
 }
 
-function RepoItem(props: {
-  data: TRepoSearchResultItem;
-  likeCount: number;
-  onLikeClick: () => void;
-}) {
-  const { data, likeCount, onLikeClick } = props;
-
-  const formattedCreatedAt = formatRelativeTime(data.created_at);
+function RepoItem(props: { data: TRepoSearchResultItem }) {
+  const { data } = props;
 
   return (
     <div>
       <h2 className="font-medium">
-        <a className="hover:underline" href={data.html_url}>
+        <a
+          className="hover:underline"
+          href={data.html_url}
+          rel="noopener"
+          target="_blank" // Open in new tab
+        >
           {data.full_name}
         </a>
       </h2>
       <p>{data.description}</p>
       <p className="text-sm text-gray-600">
         {data.stargazers_count.toLocaleString("en-US")} stars | created{" "}
-        {formattedCreatedAt}
-        {" | "}
-        <button
-          className="hover:underline"
-          onClick={() => {
-            onLikeClick();
-          }}
-        >
-          like ({likeCount})
-        </button>
+        {formatRelativeTime(data.created_at)}
       </p>
     </div>
   );
