--- a/chapter-01-10-more-styling/src/root.tsx
+++ b/chapter-03-01-state-starter/src/root.tsx
@@ -1,28 +1,23 @@
-import { formatDistanceToNowStrict } from "date-fns/formatDistanceToNowStrict";
-import { enUS as localeEnUs } from "date-fns/locale/en-US";
-import repoList from "./data.json";
+import { repoList } from "./data";
+import { formatRelativeTime } from "./helpers/time";
 
 export function GitHubRoot() {
   return (
-    <div className="flex flex-col gap-6 p-8">
+    <div className="flex flex-col gap-6">
       {repoList.map((item) => {
-        // See https://date-fns.org/v2.30.0/docs/formatDistanceToNowStrict
-        const formattedCreatedAt = formatDistanceToNowStrict(
-          new Date(item.created_at), // Parse ISO 8601 date time string
-          {
-            addSuffix: true, // Add "ago"
-            locale: localeEnUs,
-          },
-        );
+        const formattedCreatedAt = formatRelativeTime(item.created_at);
 
         return (
           <div key={item.id}>
             <h2 className="font-medium">
-              <a href={item.html_url}>{item.full_name}</a>
+              <a className="hover:underline" href={item.html_url}>
+                {item.full_name}
+              </a>
             </h2>
             <p>{item.description}</p>
             <p className="text-sm text-gray-600">
-              {item.stargazers_count} stars | created {formattedCreatedAt}
+              {item.stargazers_count.toLocaleString("en-US")} stars | created{" "}
+              {formattedCreatedAt}
             </p>
           </div>
         );
